### PR TITLE
Pursue a part of #1030, so it becomes part of the next release

### PR DESCRIPTION
**Pull request by skinkie**
## 📒 Description

Considering you are writing a very big tree, you don't want to materialise that tree before it ends up in a file in a list first. Ideally the entire subtree should only be rendered just in time.

## 🔗 What I've Done

I have allowed a Generator to be handled as a List.

## 💬 Comments

This issue does not resolve the question of #1030 to make the type annotations an `Iterable`. But I want the functionality to allow a Generator as input be part of the next release so we can use the stock xsData.

## 🛫 Checklist

- [ ] Updated docs
- [ ] Added unit-tests
- [ ] [Sample tests](https://github.com/tefra/xsdata-samples) pass
- [ ] [W3C tests](https://github.com/tefra/xsdata-w3c-tests) pass

### CODE REVIEW

File: xsdata/formats/dataclass/serializers/mixins.py [hunk 1]
```diff
@@ -816,7 +819,7 @@ def convert_elements(
         Yields:
             An iterator of sax events.
         """
-        if collections.is_array(value):
+        if collections.is_array(value) or isinstance(value, Generator):
```
Comment (tefra) [inline xsdata/formats/dataclass/serializers/mixins.py:822]: Let's move this check inside the collections.is_array util method please!

Comment (skinkie) [reply xsdata/formats/dataclass/serializers/mixins.py]: Done!

**Review by tefra — APPROVED** [review]
Nice!

**Comment by skinkie** [comment]
You made my Friday 13th well ;-)

**Pull request merged**